streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorRecordContext.java

```diff
@@ -79,26 +84,128 @@ public Headers headers() {
         return headers;
     }
 
-    public long sizeBytes() {
-        long size = 0L;
-        size += 8; // value.context.timestamp
-        size += 8; // value.context.offset
+    public int sizeBytes() {
+        int size = 0;
+        size += Long.BYTES; // value.context.timestamp
+        size += Long.BYTES; // value.context.offset
         if (topic != null) {
-            size += topic.toCharArray().length;
+            size += topic.getBytes(UTF_8).length;
         }
-        size += 4; // partition
+        size += Integer.BYTES; // partition
         if (headers != null) {
             for (final Header header : headers) {
-                size += header.key().toCharArray().length;
+                size += header.key().getBytes(UTF_8).length;
                 final byte[] value = header.value();
                 if (value != null) {
                     size += value.length;
                 }
             }
         }
+        if (size < 0) {
```

Comment:
Is there a way to check against wrapping all the way around to positive again? Even a long could wrap around in principle... This is starting to get a little complicated. Maybe we should just fall back on the jvm to catch us when we try to make and populate the byte array after all.

Reply:
> Is there a way to check against wrapping all the way around to positive again?

I think you would need to check after each modification... This would of course become costly.

> Even a long could wrap around in principle... 

True. But I think the risk would be minimal in practice for this case.

> Maybe we should just fall back on the jvm to catch us when we try to make and populate the byte array after all.

Not sure what you mean by this?

Reply:
I meant that maybe we should just remove the check.

Reply:
> or not check at all.

Fine with we to remove the check.